arm/proofs/instruction.ml

```diff
@@ -1187,6 +1187,31 @@ let arm_UMULH = define
         let d:N word = word((val n * val m) DIV (2 EXP dimindex(:N))) in
         (Rd := d) s`;;
 
+let arm_USRA_VEC = define
+ `arm_USRA_VEC Rd Rn shift esize datasize =
+    \s. let n:(128)word = read Rn s in
+        let n:(128)word =
+          if esize = 64 then usimd2 (\x. word_ushr x shift) n
+          else if esize = 32 then usimd4 (\x. word_ushr x shift) n
+          else if esize = 16 then usimd8 (\x. word_ushr x shift) n
+          else usimd16 (\x. word_ushr x shift) n in
+        let d:(128)word = read Rd s in
+        if datasize = 128 then
```

Comment:
I think I can remove this if-else branch that is based on `datasize`; I will make this PR as ready after it is done.